NN_chatCmd.js

```diff
@@ -99,33 +111,56 @@ nerthus.chatCmd.map["map"] = function(ch)
 {
     var map_url = ch.t.split(" ").slice(1).join(" ");
     map_url = nerthus.chatCmd.extractUrlFromDecorator(map_url);
-    $("#ground").css("backgroundImage","url(" + map_url + ")")
+    var map_container = $("#_ng-map-" + nerthus.chatCmd.currentMap);
+    if (map_container.length)
+        map_container.css("backgroundImage", "url(" + map_url + ")");
+    else
+    {
+        $("<img id='_ng-map-" + nerthus.chatCmd.currentMap + "'>")
+        .css({
+            "backgroundImage": "url(" + map_url + ")",
+            "width": map.x << 5,
+            "height": map.y << 5,
+            "position": "absolute"
+        }).appendTo("#ground");
+    }
     return true;
 }
 
 nerthus.chatCmd.map["light"] = function(ch)
 {
     var opacity = ch.t.split(" ")[1];
+    if (opacity === undefined)
+        opacity = "";
     $("#base").css("opacity",opacity);
     return true;
 }
 
 nerthus.chatCmd.map["addGraf"] = function(ch)
 {  //cmd[0]=x, cmd[1]=y, cmd[2]=url, cmd[3]=tip_text, cmd[4]=isCol
-    var cmd = ch.t.split(" ").slice(1).join(" ").split(",");
-    var x = parseInt(cmd[0]);
-    var y = parseInt(cmd[1]);
-    var _url = nerthus.chatCmd.extractUrlFromDecorator(cmd[2]);
-    var _tip = cmd[3] ? ' tip="<b>'+cmd[3]+'</b>" ctip="t_npc"' : "";
-    var isCol = parseInt(cmd[4]);
-    $('<img id="_ng-' + cmd[0] + '-' + cmd[1] + '" src="'+ _url +'"'+_tip+'>').css("position","absolute").appendTo('#base')
-    .load(function()
-    {  //wyśrodkowanie w osi x i wyrównanie do stóp w osi y
-        var _x = 32 * x + 16 - Math.floor($(this).width() / 2);
-        var _y = 32 * y + 32 - $(this).height();
-        $(this).css({top:"" + _y + "px", left: "" + _x + "px"}).css("z-index", y * 2 + 9);
-    })
-    if(isCol) g.npccol[ x + 256 * y] = true;
+    var map_ts = sessionStorage.getItem(ch.ts); //Test for 1st "unnamed" map
```

Comment:
Co leci w tym ch.ts? Bo ten warunek w if'ie średnio mi się podoba i pewnie można to zrobić inaczej.

Reply:
W ch.ts leci timestamp tego kiedy została wysłana wiadomość, więc jest unikalny na każdą wiadomość.